### PR TITLE
feat(datepicker): add footerTemplate property

### LINKED ISSUE #1197: DatePicker custom footer template
### Issue description:
There used to be a nice feature to turn a button group under the calendar for selecting Today, clearing the current value or closing the calendar control, which was useful when working with an input. Is there any possibility of these buttons being re-added? Is there a temporary work around to add these myself?

### Version of Angular, ng-bootstrap, and Bootstrap:
Angular: 2.3.1
ng-bootstrap: 1.0.0-alpha.16
Bootstrap: 4.0.0-alpha.5

### PR DESCRIPTION
**Pull request by dzonatan**
Closes #1197

Not sure about `@since 3.3.0` flag. Let me know.

### CODE REVIEW

File: demo/src/app/components/datepicker/demos/footertemplate/datepicker-footertemplate.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      <div class="input-group-append">
+        <button class="btn btn-outline-secondary calendar" (click)="d.toggle()" type="button"></button>
+      </div>
+    </div>
+  </div>
+</form>
+
+<ng-template #footerTemplate let-datepicker>
+  <hr class="my-0">
+  <button class="btn btn-primary btn-sm m-2 float-left" (click)="datepicker.onDateSelect(today)">Today</button>
```
Comment (maxokorokov) [inline demo/src/app/components/datepicker/demos/footertemplate/datepicker-footertemplate.html]: Please change to `model = today; d.close()` for now

File: demo/src/app/components/datepicker/demos/footertemplate/datepicker-footertemplate.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<p>This datepicker uses a footer template which is presented inside datepicker. Today and close buttons used as an example.</p>
+
+<form class="form-inline">
+  <div class="form-group">
+    <div class="input-group">
+      <input class="form-control" placeholder="yyyy-mm-dd" name="dp" ngbDatepicker [footerTemplate]="footerTemplate" #d="ngbDatepicker">
```
Comment (maxokorokov) [inline demo/src/app/components/datepicker/demos/footertemplate/datepicker-footertemplate.html]: please add `[(ngModel)] = "model"` and see comment below

File: demo/src/app/components/datepicker/overview/datepicker-overview.component.ts [hunk 3]
```diff
@@ -73,6 +73,13 @@ providers: [{provide: NgbDateParserFormatter, useClass: YourOwnParserFormatter}]
 </ng-template>
 
 <ngbDatepicker [dayTemplate]=“t”/>
+`,
+    footerTemplate: `
+<ng-template #t let-datepicker>
+  <button (click)="datepicker.onDateSelect(today)">Today</button>
```
Comment (maxokorokov) [inline demo/src/app/components/datepicker/overview/datepicker-overview.component.ts]: again should be `model = today` or `d.close()` as and example, not `onDateSelect()`

File: src/datepicker/datepicker-footer-template-context.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import {NgbDatepicker} from './datepicker';
+
+/**
+ * Context for the datepicker 'footer' template in case you want to override the default one
+ */
+export interface FooterTemplateContext {
```
Comment (maxokorokov) [inline src/datepicker/datepicker-footer-template-context.ts]: Let's drop this context altogether (see main comment)

File: src/datepicker/datepicker.ts [hunk 5]
```diff
@@ -100,6 +101,8 @@ export interface NgbDatepickerNavigateEvent {
         </div>
       </ng-template>
     </div>
+
+    <ng-container *ngTemplateOutlet="footerTemplate; context: { $implicit: datepicker, datepicker: datepicker }"></ng-container>
```
Comment (maxokorokov) [inline src/datepicker/datepicker.ts]: Let's replace this with `<ng-template [ngTemplateOutlet]></ng-template>` form and drop context

File: src/datepicker/datepicker.ts [hunk 6]
```diff
@@ -133,6 +136,13 @@ export class NgbDatepicker implements OnDestroy,
    */
   @Input() firstDayOfWeek: number;
 
+  /**
+   * Reference for the custom template for the footer
+   *
+   * @since 3.3.0
+   */
+  @Input() footerTemplate: TemplateRef<FooterTemplateContext>;
```
Comment (maxokorokov) [inline src/datepicker/datepicker.ts]: `TemplateRef<{}>` I guess...

File: src/datepicker/datepicker.ts [hunk 7]
```diff
@@ -194,13 +204,15 @@ export class NgbDatepicker implements OnDestroy,
   onChange = (_: any) => {};
   onTouched = () => {};
 
+  get datepicker() { return this; }
```
Comment (maxokorokov) [inline src/datepicker/datepicker.ts]: to remove

File: src/datepicker/datepicker.ts (hunk 6 above)
Comment (dzonatan) [reply src/datepicker/datepicker.ts]: To be consistent with other places `TemplateRef<any>` was used.

**Comment by maxokorokov** [comment]
LGTM, thanks!

**Pull request closed without merging**